source/git-promise.js

```diff
@@ -282,34 +282,30 @@ git.binaryFileContent = (repoPath, filename, version, outPipe) => {
 }
 
 git.diffFile = (repoPath, filename, sha1, ignoreWhiteSpace) => {
-  const newFileDiffArgs = ['diff', '--no-index', isWindows ? 'NUL' : '/dev/null', filename.trim()];
+  if (sha1) {
+    return git(['rev-list', '--max-parents=0', sha1], repoPath).then((initialCommitSha1) => {
+      let prevSha1 = sha1 == initialCommitSha1.trim() ? '4b825dc642cb6eb9a060e54bf8d69288fbee4904' : `${sha1}^`;
```

Comment:
"empty tree" commit sha1. See https://stackoverflow.com/a/40884093/9663233

Reply:
Could you extract that sha into a constant and add this comment there too?